boards/FSM/Src/App/App_WheelSpeedSensor.c

```diff
@@ -0,0 +1,44 @@
+#include <stdlib.h>
+#include <assert.h>
+#include <stddef.h>
+#include <math.h>
+#include "App_WheelSpeedSensor.h"
+
+struct WheelSpeedSensor
+{
+    float (*get_wheel_speed)(void);
+};
+
+struct WheelSpeedSensor *
+    App_WheelSpeedSensor_Create(float (*get_wheel_speed)(void))
+{
+    assert(get_wheel_speed != NULL);
+    struct WheelSpeedSensor *wheel_speed_sensor =
+        (struct WheelSpeedSensor *)malloc(sizeof(struct WheelSpeedSensor));
+
+    assert(wheel_speed_sensor != NULL);
+
+    wheel_speed_sensor->get_wheel_speed = get_wheel_speed;
+    return wheel_speed_sensor;
+}
+
+void App_WheelSpeedSensor_Destroy(
+    struct WheelSpeedSensor *const wheel_speed_sensor)
+{
+    free(wheel_speed_sensor);
+}
+
+float App_WheelSpeedSensor_GetWheelSpeed(
+    struct WheelSpeedSensor *const wheel_speed_sensor)
+{
+    return wheel_speed_sensor->get_wheel_speed();
+}
+
+enum WheelSpeedFaultStatus App_WheelSpeedSensor_GetFaultStatus(
```

Comment:
I would remove this entirely, we should just check that the wheel speed > 150 and set a non critical fault in the state machine.

Reply:
```
driveStateRunOnTick() {
wheel_speed = get_wheel_speed();
if (wheel_speed > 150.0f) 
  set_non_crit_fault();
}
```

Reply:
And GetWheelSpeed should just return an exit code

```
ExitCode GetWheelSpeed(struct Wheel* wheel, float* returned_wheel_speed) {
    const float wheel_speed = wheel->get_wheel_speed();
    ExitCode exit_code = EXIT_CODE_OK;

   if (isnanf(wheel_speed))
   {
      exit_code = EXIT_CODE_OUT_OF_RANGE;
   }
    *returned_wheel_speed = wheel_speed;
   return exit_code;
}
```

Reply:
Honestly since we're fine with transmitting NAN over CAN. We probably don't even need the `isnanf` check.

Reply:
Yeah I ended up just going with setting the fault high when you go above 150 km/h

Reply:
Do you still prefer having `GetWheelSpeed` return an ExitCode? Rn my implementation is similar to https://github.com/UBCFormulaElectric/Consolidated-Firmware/pull/582#discussion_r417060913 and I don't see how returning ExitCode will be helpful in this case

Reply:
Yeah maybe we can just not return anything. Or we could treat negative values as a faulty value since our math shouldn't ever return negative speed. Your call, I'm OK with either option.